src/redux/contacts/reducer.ts

```diff
@@ -96,33 +96,34 @@ const contactReducer = createReducer(initialState, (contacts) => {
       })
     )
     .addCase(
-      ContactActions.setNewContact.pending,
+      ContactActions.saveNewContact.pending,
       (state: ContactsState) => ({
         ...state,
         loading: true
       })
     )
     .addCase(
-      ContactActions.setNewContact.fulfilled,
+      ContactActions.saveNewContact.fulfilled,
       (state: ContactsState, { payload }) => ({
         ...state,
         loading: false,
-        newContact: payload,
+        newContact: {} as Contact,
+        contacts: [...state.contacts, payload],

```

Comment:
Here is where I tried to implement your suggestion, but it seems the value is `NULL`. I also tried using `payload.data` instead of just `payload` but that threw an error not recognizing `data` on `payload`, if that makes sense.